components/sync15/src/request.rs

```diff
@@ -331,57 +338,61 @@ where
             serde_json::to_writer(&mut self.queued, &record).unwrap();
         }
 
+        // Add the record to the queue tracker
         self.post_limits.record_added(payload_length);
         self.batch_limits.record_added(payload_length);
 
         Ok(true)
     }
 
+
+    /// Returns batchid 
+    #[inline]
+    fn batch_id(&self) -> String {
+        //Return current batch_id or "true", if not already in batch
+        match &self.batch {
+            BatchState::NewBatch => "true".into(),
+            BatchState::InBatch(ref s) => s.clone(),
+        }
+    }
+
+    // Flushes the current queue
+    // want_commit??? TODO 
```

Comment:
Since this is a public function, this comment should be a doc comment.

Also, get rid of comments to yourself.